mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/MasterMain.java

```diff
@@ -204,17 +194,31 @@ public MasterMain(
             // services
             mantisServices.addService(jobClustersManagerService);
 
-            if (this.config.isLocalMode()) {
-                mantisServices.addService(new MasterApiAkkaService(new LocalMasterMonitor(leadershipManager.getDescription()), leadershipManager.getDescription(), jobClusterManagerActor, statusEventBrokerActor,
-                       resourceClusters, resourceClustersHostActor, config.getApiPort(), storageProvider, lifecycleEventPublisher, leadershipManager));
-                leadershipManager.becomeLeader();
+            // set up leader election
+            final ILeaderElectorFactory leaderFactory;
+            final MasterMonitor monitor;
+            if(!config.isLocalMode() && config.getLeaderElectorFactory() instanceof LocalLeaderFactory) {
```

Comment:
Critical injection point is here. The if/else is only necessary because I wanted to support backwards compatibility with configs that exist today. I thought if I broke configs as they are today we would need a major version bump. I tested with the old config and then modified the properties files in the container test to make sure it worked. I left the source code updated for the new property configuration.